Cesium3DTilesSelection/src/Tile.cpp

```diff
@@ -662,16 +662,26 @@ getTileBoundingRegionForUpsampling(const Tile& parent) {
     assert(!details.rasterOverlayRectangles.empty());
 
     // Use the projected center of the tile as the subdivision center.
-    // The tile will be subdivided by (0.5, 0.5) in the _first_ overlay's
-    // texture coordinates (no matter which overlay(s) had more detail), at
-    // least until https://github.com/CesiumGS/cesium-native/issues/385 is
-    // addressed. So pick the center accordingly.
-    glm::dvec2 centerProjected = details.rasterOverlayRectangles[0].getCenter();
-    Cartographic center = unprojectPosition(
-        details.rasterOverlayProjections[0],
-        glm::dvec3(centerProjected, 0.0));
-
-    return RegionAndCenter{details.boundingRegion, center};
+    // The tile will be subdivided by (0.5, 0.5) in the first overlay's
+    // texture coordinates which overlay had more detail.
+
+    for (size_t i = 0; i < parent.getMappedRasterTiles().size(); ++i) {
+      RasterMappedTo3DTile* pMapped =
+          const_cast<RasterMappedTo3DTile*>(&parent.getMappedRasterTiles()[i]);
+      if (pMapped->update(*const_cast<Tile*>(&parent)) ==
```

Comment:
A method called `getTileBoundingRegionForUpsampling` shouldn't have a side-effect where it updates the raster overlay mappings. The need for the const cast reinforces that. Is it needed? It looks to me like this method is always called after all the RasterOveralyTile's isMoreDetailAvailable field is updated, so we can avoid the side effect and some unnecessary work by using it.

Reply:
Thanks, I called Update out of convenience only to know whether the tile had more detail. I added a lightweight const-safe function that does just that.